src/main/java/stirling/software/SPDF/controller/api/misc/DecompressPdfController.java

```diff
@@ -0,0 +1,137 @@
+package stirling.software.SPDF.controller.api.misc;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.pdfbox.Loader;
+import org.apache.pdfbox.cos.*;
+import org.apache.pdfbox.io.IOUtils;
+import org.apache.pdfbox.pdfwriter.compress.CompressParameters;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+import lombok.extern.slf4j.Slf4j;
+
+import stirling.software.SPDF.model.api.PDFFile;
+import stirling.software.SPDF.utils.WebResponseUtils;
+
+@RestController
+@RequestMapping("/api/v1/misc")
+@Slf4j
+@Tag(name = "Misc", description = "Miscellaneous APIs")
+public class DecompressPdfController {
+
+    @PostMapping(value = "/decompress-pdf", consumes = "multipart/form-data")
+    @Operation(
+            summary = "Decompress PDF streams",
+            description = "Fully decompresses all PDF streams including text content")
+    public ResponseEntity<byte[]> decompressPdf(@ModelAttribute PDFFile request)
+            throws IOException {
+
+        MultipartFile file = request.getFileInput();
+
+        try (PDDocument document = Loader.loadPDF(file.getBytes())) {
+            // Process all objects in document
+            processAllObjects(document);
+
+            // Save with explicit no compression
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            document.save(baos, CompressParameters.NO_COMPRESSION);
+
+            String outputFilename =
+                    file.getOriginalFilename().replaceFirst("\\.(?=[^.]+$)", "_decompressed.");
+            return WebResponseUtils.bytesToWebResponse(
+                    baos.toByteArray(), outputFilename, MediaType.APPLICATION_PDF);
+        }
+    }
+
+    private void processAllObjects(PDDocument document) {
+        Set<COSBase> processed = new HashSet<>();
+        COSDocument cosDoc = document.getDocument();
+
+        // Process all objects in the document
+        for (COSObjectKey key : cosDoc.getXrefTable().keySet()) {
```

Comment:
Could also do this operation using lambdas:
```suggestion
        cosDoc.getXrefTable()
            .keySet()
            .stream()
            .filter(key -> !cosDoc.getObjectFromPool(key).isObjectNull())
            .filter(key -> !processed.contains(cosDoc.getObjectFromPool(key)))
            .forEach(key -> {
                COSObject obj = cosDoc.getObjectFromPool(key);
                processed.add(obj);
                processObject(obj, processed);
            });
```
And then you can remove the null/contains check in the `processObject()` method